src/ClipboardCopy.js

```diff
@@ -37,11 +50,27 @@ export default class ClipboardCopy extends React.Component {
 
   render() {
     // eslint-disable-next-line no-unused-vars
-    const {value = '', ...rest} = this.props
-    return (
-      <Button onClick={() => this.copy()} type="button" {...rest}>
-        <Octicon icon={Clippy} />
-      </Button>
+    const {value = '', as, ...rest} = this.props
+
+    const Icon = (
+      <StyledOcticon icon={this.state.copied ? Check : Clippy} color={this.state.copied ? 'green.5' : 'inherit'} />
     )
+
+    if (as === 'link')
+      return (
+        <Link onClick={this.onClick.bind(this)} style={{cursor: 'pointer'}} {...rest}>
+          {Icon}
+        </Link>
+      )
+    else
+      return (
+        <Button onClick={this.onClick.bind(this)} type="button" {...rest}>
+          {Icon}
+        </Button>
+      )
```

Comment:
I know @colebemis prefers to skip the `else` keyword. I kind of like it better this way 😇

Happy to go either way.

Reply:
I'm kindddd of thinking that I'd like to _only_ render/style this as a `Button`. We try to shy away fro having too many variations in one component, and in this case I think it would be better to just have one style for Copy to Clipboard across the board. Would you mind removing the `link` prop and the if/else statement? Sorry, I should have caught this in your last PR!

Reply:
Unless you think the Button looks to weird with the check animation? What do you think?

Reply:
Either way, my feeling is this should either be an unstyled `<button>` or a regular `Button`, but we should only have one option :)

Reply:
I agree! Big fan of reducing the number of ways to do the same thing 👍

Right now, there is a linky looking copy button on .com. Do you feel this should be 
- brought into primer-components and improved on (example: it could use better a aria-label), 
or 
- pattern should be deprecated and this button should sit with the other ones on the right (i like the latter)

![Screenshot 2019-05-28 at 8](https://user-images.githubusercontent.com/1863771/58491227-3ff5f900-818c-11e9-8615-ac651f05b6d5.png)

Reply:
Oops, that took a while! All good now